### ISSUE TITLE
lambda function does not accepted by :bind keyword

### ISSUE THREAD

**Issue by conao3** [conao3/leaf.el#322]
## Description
lambda function does not accepted by :bind keyword

## Issue leaf-block
```elisp
(leaf custom
  :bind ("M-ESC ESC" . (lambda () (interactive) (redraw-frame)))) 
```

## Expected output from macroexpand-1 of the leaf-block
```elisp
(prog1 'custom
  (leaf-keys
    (("M-ESC ESC" . (lambda () (interactive) (redraw-frame))))))
```

## Actual output
```
Debugger entered--Lisp error: (wrong-type-argument listp "M-ESC ESC")
  car("M-ESC ESC")
  (listp (car bind))
  (and (listp (car bind)) (funcall pairp (car bind)))
```

**Comment by nasyxx** [comment]
Hi, why don't you directly use the `leaf-pairp` instead of a `let pairp`

https://github.com/conao3/leaf.el/blob/fbe9bfba5ef4c99ee190b259a5e3341dd7a71935/leaf.el#L477-L485

https://github.com/conao3/leaf.el/blob/fbe9bfba5ef4c99ee190b259a5e3341dd7a71935/leaf.el#L839-L862

which causes the `listp` error here?

**Comment by conao3** [comment]
It just `leaf-pairp` for macro. So, it needs `eval` in macro.

**Comment by nasyxx** [comment]
I see.  So, is there still no good step to solve this problem?

**Comment by conao3** [comment]
Yes. I recommend define function in `:preface` section, then bind some keybind.
However, any contribution is welcome. I'm now finding some work, I'm a bit busy...
